File: src/tf_broadcaster_camera.py
#!/usr/bin/env python
'''raspi_camera TF'''
import os
import roslib
import rospy
import tf
roslib.load_manifest('myrobot')


def talker():
    '''camera tf'''
    rospy.init_node('raspicam_tf', anonymous=True)
    br = tf.TransformBroadcaster()
    rate = rospy.Rate(10)  # 10hz
    while not rospy.is_shutdown():
        br.sendTransform((0.05, 0.0, 0.15),
                         (0.0, 0.0, 0.0, 1.0),
                         rospy.Time.now(), "raspicam", "base_link")
        rate.sleep()


if __name__ == '__main__':
    try:
        talker()
    except rospy.ROSInterruptException:
        pass

File: src/tf_broadcaster_camera.cpp
#include <ros/ros.h>
#include <tf/transform_broadcaster.h>

int main( int argc, char* argv[] )
{
    ros::init( argc, argv, "camera_broadcaster" );
    ros::NodeHandle nh;

    tf::TransformBroadcaster br;
    tf::Transform transform;

    ros::Rate rate( 10.0 );

    while ( nh.ok() )
    {
        transform.setOrigin( tf::Vector3( 0.05, 0.0, -0.15 ) );
        transform.setRotation( tf::Quaternion( 0, 0, 0, 1 ) );
        br.sendTransform( tf::StampedTransform( transform, ros::Time::now(), "raspicam", "odom" ) );

        rate.sleep();
    }

    return 0;
}
